virtManager/details/snapshots.py

```diff
@@ -108,9 +110,48 @@ def _cleanup(self):
     # UI init #
     ###########
 
+    def _init_snapshot_mode(self):
+        mode = self.widget("snapshot-new-mode")
+        model = Gtk.ListStore(str)
+
+        capsinfo = self.vm.xmlobj.lookup_capsinfo()
+
+        if capsinfo.guest.supports_externalSnapshot():
+            model.append(["external"])
+
+        model.append(["internal"])
+        mode.set_model(model)
+        uiutil.init_combo_text_column(mode, 0)
+
+    def _init_memory_path(self):
+        mempaths = self.widget("snapshot-new-memory-path")
+        model = Gtk.ListStore(str)
+        mempaths.set_model(model)
+        uiutil.init_combo_text_column(mempaths, 0)
+
     def _init_ui(self):
         buf = Gtk.TextBuffer()
         self.widget("snapshot-new-description").set_buffer(buf)
+        self._init_snapshot_mode()
+        self._init_memory_path()
+
+    def _populate_memory_path(self):
+        mempaths = self.widget("snapshot-new-memory-path")
+        model = mempaths.get_model()
+        paths = set()
+
+        model.clear()
+
+        for disk in self.vm.get_xmlobj().devices.disk:
+            dirpath = os.path.dirname(disk.get_source_path())
```

Comment:
It could be interesting to users having VMs with multiple disks located in different places on the host system. I was considering adding code to figure out boot order and sort the paths based on that but eventually decided it's not worth the effort.

I'll probably changes the way how we tell libvirt to make external snapshot. For running VM adding path to memory state should be enough and for offline VM using disk-only is enough as well. So we will no longer hardcode any disk to the snapshot XML.